File: bfs.cs
using System;
using System.Collections.Generic;
					
public class Program
{
	public static void Main()
	{
        var tree = new Dictionary<int, List<int>> {
            {10, new List<int>{8, 13}},
            {8, new List<int>{3, 9}},
            {13, new List<int>{15}},
            {3, new List<int>()},
            {9, new List<int>()},
            {15, new List<int>()},
        };
		
        Bfs(tree, 10, 9);
	}
		
	public static int Bfs(Dictionary<int, List<int>> tree, int rootNode, int goal)
	{
		var visited = new List<int>();
		var queue = new Queue<int>();
		
		visited.Add(rootNode);
		queue.Enqueue(rootNode);

		while(queue.Count > 0)
        {
			var node = queue.Dequeue();
			Console.WriteLine(node.ToString() + " ");
			if (node == goal)
			{            
				Console.WriteLine("\nFound goal");
				return node;
			}

			foreach(var neighbor in tree[node])
            {
				if (!visited.Contains(neighbor))
                {
					visited.Add(neighbor);
					queue.Enqueue(neighbor);
                }
            }
        }
		return -1;
	}
}

File: bfs.py
tree = {
    10 : [8, 13],
    8 : [3, 9],
    13 : [15],
    3 : [],
    9 : [],
    15 : []
}


visited, queue = [], []


def bfs(root_node, goal):
    visited.append(root_node)
    queue.append(root_node)

    while queue:
        node = queue.pop(0)
        print(node, end=' ')
        if node == goal:            
            print('\nFound goal')
            return node

        for neighbor in tree[node]:
            if neighbor not in visited:
                visited.append(neighbor)
                queue.append(neighbor)

bfs(root_node=10, goal=1)
